torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -939,6 +940,47 @@ def test_new_subgroups_overlap_not_allowed(self):
                     ranks_per_subgroup_list=[[0], [1, 2], [1, 3]]
                 )
 
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "MPI backend does not support creating subgroups on CUDA devices",
+        )
+        @skip_if_lt_x_gpu(2)
+        def test_average_parameters(self):
+            rank = dist.get_rank()
+            rank_to_GPU = self._init_multigpu_helper()
+            device_id = rank_to_GPU[rank][0]
+
+            model = (
+                nn.Sequential(
+                    nn.Conv2d(3, 3, kernel_size=3, padding=1),
+                    nn.ReLU(),
+                    nn.Linear(1, 5, bias=False)
+                ).cuda(device_id)
+            )
+
+            # Test global model averaging
+            for p in model.parameters():
+                p.data = torch.ones_like(p.data)
+            model_averaging_utils.average_parameters(module=model, process_group=None)
+            # Every element will be the same as the input.
+            for p in model.parameters():
+                self.assertEqual(p.data, torch.ones_like(p.data))
+
+            # Test partial model averaging
+            for p in model.parameters():
+                p.data = torch.ones_like(p.data) * rank
+            group_nccl = dist.new_group(ranks=[0, 1], backend="nccl")
+            model_averaging_utils.average_parameters(module=model, process_group=group_nccl)
+            if not dist._rank_not_in_group(group_nccl):
+                # Every element on device 0 or 1 should be the average of 0 and 1, i.e., 0.5.
+                for p in model.parameters():
+                    self.assertEqual(p.data, torch.ones_like(p.data) * 0.5)
+            else:
+                # Every element on device not in the subgroup should be -rank,
+                # because `get_world_size()` returns -1.
+                for p in model.parameters():
+                    self.assertEqual(p.data, torch.ones_like(p.data) * -rank)
```

Comment:
Ideally, non group members should not have their parameters modified right? can we just check `_rank_not_in_group` in the averaging function and early return if the rank isn't in group, so the params are not touched?

Reply:
Yeah, no update should be better. An early return now is provided for non-group members.